nixos/modules/system/boot/plymouth.nix

```diff
@@ -146,6 +148,7 @@ in
     # We use `mkAfter` to ensure that LUKS password prompt would be shown earlier than the splash screen.
     boot.initrd.preLVMCommands = mkAfter ''
       mkdir -p /etc/plymouth
+      mkdir /run/plymouth
```

Comment:
Is there any concern around this existing / should we make it `-p` just in case?

Reply:
There's definitely no harm in `-p` but it's definitely not necessary. `/run` is created extremely early in stage 1.